chisel/src/executor.rs

```diff
@@ -0,0 +1,766 @@
+                {
+                    // These *should* never panic after a successful compilation.
+                    let bytecode =
+                        contract.get_bytecode_bytes().expect("No bytecode for contract.");
+                    let deployed_bytecode = contract
+                        .get_deployed_bytecode_bytes()
+                        .expect("No deployed bytecode for contract.");
+
+                    // Fetch the run function's body statement
+                    let run_func_statements = compiled.intermediate.run_func_body()?;
+
+                    // Find the last statement within the "run()" method and get the program
+                    // counter via the source map.
+                    if let Some(final_statement) = run_func_statements.last() {
+                        let final_pc = {
+                            let source_loc = final_statement.loc();
+                            let offset = source_loc.start();
+                            let length = source_loc.end() - source_loc.start();
+                            contract
+                                .get_source_map_deployed()
+                                .unwrap()
+                                .unwrap()
+                                .into_iter()
+                                .zip(InstructionIter::new(&deployed_bytecode))
+                                .filter(|(s, _)| s.offset == offset && s.length == length)
+                                .map(|(_, i)| i.pc)
+                                .max()
+                                .unwrap_or_default()
+                        };
+
+                        // Create a new runner
+                        let mut runner = self.prepare_runner(final_pc).await;
+
+                        // Return [ChiselResult] or bubble up error
+                        match runner.run(bytecode.into_owned()) {
+                            Ok(res) => Ok(res),
+                            Err(e) => Err(e),
+                        }
+                    } else {
+                        // Return a default result if no statements are present.
+                        Ok((Address::zero(), ChiselResult::default()))
+                    }
+                } else {
+                    eyre::bail!("Failed to find REPL contract!")
+                }
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Inspect a contract element inside of the current session
+    ///
+    /// ### Takes
+    ///
+    /// A solidity snippet
+    ///
+    /// ### Returns
+    ///
+    /// If successful, a formatted inspection output.
+    /// If unsuccessful but valid source, `Some(None)`
+    /// If unsuccessful, `Err(e)`
+    pub async fn inspect(&mut self, item: &str) -> Result<Option<String>> {
+        match self.clone_with_new_line(format!("bytes memory inspectoor = abi.encode({item})")) {
+            Ok((mut source, _)) => match source.execute().await {
+                Ok((_, res)) => {
```

Comment:
I guess we can get rid of a couple of levels by using `let .. = if let Ok() else {return}` 

this would make it more readable.

Reply:
Fixed